docs/data-sources/git_repository.md

```diff
@@ -0,0 +1,145 @@
+---
+Title: "Git Repository Data Source"
+Description: |-
+    Creating the Git Repository Data Source.
+---
+
+# Git Repository
+
+The `tanzu-mission-control_git_repository` Data Source allows you to get git repository to a particular scope through Tanzu Mission Control.
+
+[git-repository]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-26C2D2F3-0E5C-4E56-B875-B7FB003267E4.html
+
+## git-repository Scope
+
+In the Tanzu Mission Control resource hierarchy, there are two levels at which you can specify git repository resources:
+- **object groups** - `cluster_group` block under `scope` sub-resource
+- **Kubernetes objects** - `cluster` block under `scope` sub-resource
+
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script for the provider to function.
+Only one scope per resource is allowed.
+
+## Cluster group scoped Git Repository
+
+### Example Usage
+
+```terraform
+# Read Tanzu Mission Control git repository : fetch cluster group git repository details
```

Comment:
Regenerate the docs post updating the examples.

Reply:
sure